Zend/zend_vm_def.h

```diff
@@ -6188,7 +6188,7 @@ ZEND_VM_HANDLER(77, ZEND_FE_RESET_R, CONST|TMP|VAR|CV, JMP_ADDR)
 			}
 		}
 	} else {
-		zend_error(E_WARNING, "Invalid argument supplied for foreach()");
+		zend_error(E_WARNING, "Argument supplied for foreach() must be of type iterable, %s given", zend_get_type_by_const(Z_TYPE_P(array_ptr)));
```

Comment:
This is technically not quite right, because you can also iterate over non-Traversable objects. But maybe it's better to tell a while lie ;)

Reply:
Educational-purpose lie? :D But probably the `array|object` type is more straightforward (and doesn't even lie)